composer/core/data_spec.py

```diff
@@ -117,12 +124,36 @@ def _split_mapping(m, microbatch_size: int):
     for k, v in m.items():
         if isinstance(v, torch.Tensor):
             chunked[k] = _split_tensor(v, microbatch_size)
-        if isinstance(v, (List, Tuple)):
+        elif isinstance(v, (List, Tuple)):
             chunked[k] = _split_list(v, microbatch_size)
-    num_chunks = len(list(chunked.values())[0])
+        elif isinstance(v, (int, float, str, bool)):
+            # Defer broadcasting primitives until we know num_chunks
+            pass
+        else:
+            raise ValueError(f'Unsupported batch type: {type(v)}.')
+    num_chunks = 1  # Default to 1 chunks if there are no tensors or everything is primitive
+    if len(chunked.keys()) != 0:
+        num_chunks = len(list(chunked.values())[0])
```

Comment:
```suggestion
    if len(chunked.keys()) != 0:
        num_chunks = len(list(chunked.values())[0])
    else:
        num_chunks = 1  # Default to 1 chunks if there are no tensors or everything is primitive
```

Reply:
prefer as is, explicitly indicates scope of `num_chunks` as outside if/else

Reply:
but the readability... ok fine